openmc/weight_windows.py

```diff
@@ -410,3 +411,182 @@ def from_hdf5(cls, group, meshes):
             weight_cutoff=weight_cutoff,
             id=id
         )
+
+def __wwinp_reader(path):
+    """
+    Generator that returns the next value in a wwinp file.
+
+    path : str or pathlib.Path
+        Location of the wwinp file
+    """
+
+    with open(path, 'r') as fh:
+
+        # read the first line of the file and
+        # keep only the first four entries
+        while True:
+            line = next(fh)
+            if line and not line.startswith('c'):
+                break
+
+        values = line.strip().split()[:4]
+        for value in values:
+            yield value
+
+        # the remainder of the file can be read as
+        # sequential values
+        while True:
+            line = next(fh)
+            # skip empty or commented lines
+            if not line or line.startswith('c'):
+                continue
+            values = line.strip().split()
+            for value in values:
+                yield value
+
+
+def wwinp_to_wws(path):
+    """Creates WeightWindows classes from a wwinp file
+
+    Parameters
+    ----------
+    path : str or pathlib.Path object
+        Path to the wwinp file
+
+    Returns
+    -------
+    list of openmc.WeightWindows
+    """
+    # create generator for getting the next parameter from the file
+    wwinp = __wwinp_reader(path)
+
+    # first parameter, if, of wwinp file is unused
+    next(wwinp)
+
+    # check time parameter, iv
+    if int(float(next(wwinp))) > 1:
+        raise ValueError('Time-dependent weight windows '
+                         'are not yet supported.')
+
+    # number of particle types, ni
+    n_particle_types = int(float(next(wwinp)))
+
+    # read an indicator of the mesh type.
+    # this will be 10 if a rectilinear mesh
+    # and 16 for cylindrical or spherical meshes
+    mesh_chars = int(float(next(wwinp)))
+
+    if mesh_chars != 10:
+        # TODO: read the first entry by default and display a warning
+        raise NotImplementedError('Cylindrical and Spherical meshes '
+                                  'are not currently supported')
+
+    # read the number of energy groups for each particle, ne
+    n_egroups = [int(next(wwinp)) for _ in range(n_particle_types)]
```

Comment:
I would just add in the logic for reading time bins before this even though we're not supporting that yet.

Reply:
I'm game for that. How would you recommend we handle associating the outgoing `WeightWindow` classes with their time bins? Ideally, this would be an attribute of the class we can set and use in simulation, but we aren't quite there yet.

Reply:
Basically what I meant is what @eepeterson suggested above -- we would have the logic in place for reading the time bins, but we wouldn't actually do anything with them

Reply:
Ah, gotcha. Makes sense. Thanks!